### PR TITLE
DT-448 Migrate Gnome Dictionary using Gnome42/Core22

### PR DESCRIPTION
**Pull request by sergio-costas**
# Pull Request Template

## Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please check only the options that are relevant.

- [ ] General Maintenance
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration.

**Test Configuration**:

* OS (please include version):
* Any other relevant environment information:

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have checked my code and corrected any misspellings

### CODE REVIEW

File: patches/01-fix_build.diff [hunk 1]
```diff
@@ -0,0 +1,51 @@
+diff --git a/data/appdata/meson.build b/data/appdata/meson.build
```
Comment (hellsworth) [inline patches/01-fix_build.diff]: Please add a comment section at the top saying what this patch is for.

File: snapcraft.yaml [hunk 2]
```diff
@@ -32,19 +32,20 @@ parts:
...
+      - --prefix=/usr
+      - --buildtype=release
     override-pull: |
-      snapcraftctl pull
-      snapcraftctl set-version $(git describe --tags --abbrev=10)
+      craftctl default
+      craftctl set version=$(git describe --tags --abbrev=10)
       # Remove the meson.build line for erroring on redundant-declarations
       sed -i.bak -e '/.*-Werror=redundant-decls.*$/d' src/meson.build
+      patch -p1 < $CRAFT_PROJECT_DIR/patches/01-fix_build.diff
```
Comment (hellsworth) [inline snapcraft.yaml]: You could use *.diff but don't have to.

**Review by hellsworth — CHANGES_REQUESTED** [review]
Just want a comment in the patch. Otherwise looks great, thanks!

File: patches/01-fix_build.diff (hunk 1 above)
Comment (sergio-costas) [reply patches/01-fix_build.diff]: Done.

File: snapcraft.yaml (hunk 2 above)
Comment (sergio-costas) [reply snapcraft.yaml]: I prefer to set them explicitly.

**Review by hellsworth — APPROVED** [review]
Looks great thanks!

**Pull request merged**